patches/9989-misc/ps0043-p0001-ntdll-Redirect-wine_dbg_log-printf-formattin.patch

diff --git a/include/wine/debug.h b/include/wine/debug.h
index 11111111111..11111111111 100644
--- a/include/wine/debug.h
+++ b/include/wine/debug.h
@@ -346,7 +346,7 @@ static inline const char *wine_dbgstr_rect( const RECT *rect )
 static inline const char *wine_dbgstr_longlong( ULONGLONG ll )
 {
     if (sizeof(ll) > sizeof(unsigned long) && ll >> 32)
-        return wine_dbg_sprintf( "%lx%08lx", (unsigned long)(ll >> 32), (unsigned long)ll );
+        return wine_dbg_sprintf( "%x%08x", (unsigned int)(ll >> 32), (unsigned int)(ll & 0xFFFFFFFF) );
     else return wine_dbg_sprintf( "%lx", (unsigned long)ll );
 }
 
